Windows: faster-whisper インストール時に requests モジュールが不足

## 🐛 バグ報告

### 環境
- **OS**: Windows
- **バージョン**: v1.8.1
- **Python**: 3.12

### 問題の概要
KoeMoji-Goが自動的に `faster-whisper` をインストールした後、文字起こし実行時に以下のエラーが発生：

```
ModuleNotFoundError: No module named 'requests'
```

### エラーログ
```
2025/10/31 11:43:56 [INFO] FasterWhisper not found. Attempting to install...
2025/10/31 11:43:56 [INFO] Installing faster-whisper and whisper-ctranslate2...
2025/10/31 11:43:57 [INFO] FasterWhisper installed successfully
...
2025/10/31 11:43:58 [DEBUG] [STDERR] ModuleNotFoundError: No module named 'requests'
2025/10/31 11:43:58 [ERROR] 文字起こしに失敗: exit status 1
```

### 根本原因
- `pip install faster-whisper whisper-ctranslate2` 実行時に、依存パッケージの `requests` がインストールされない
- `requests` モジュールは faster-whisper の依存関係に含まれているが、Windowsの一部環境で正しくインストールされないケースがある
- pipのバージョンが古い場合、依存関係の解決に失敗する可能性

### 解決策（ユーザー側）
以下のコマンドを実行：

```cmd
# 1. pipのアップデート
python -m pip install --upgrade pip

# 2. requestsモジュールのインストール
pip install requests

# 3. faster-whisperの再インストール（推奨）
pip install --upgrade --force-reinstall faster-whisper whisper-ctranslate2
```

### 提案（開発側の改善案）

#### オプション1: インストールコマンドの改善
`internal/whisper/whisper.go` の `installFasterWhisper()` 関数を修正：

```go
func installFasterWhisper(log *log.Logger, logBuffer *[]logger.LogEntry, logMutex *sync.RWMutex) error {
    logger.LogInfo(log, logBuffer, logMutex, "Installing faster-whisper and whisper-ctranslate2...")
    
    // 1. pipのアップデート
    pipUpgradeCmd := createCommand("python", "-m", "pip", "install", "--upgrade", "pip")
    if err := pipUpgradeCmd.Run(); err != nil {
        logger.LogError(log, logBuffer, logMutex, "pip upgrade failed: %v", err)
    }
    
    // 2. 依存関係を含めた完全インストール
    cmd := createCommand("pip", "install", "--upgrade", "requests", "faster-whisper", "whisper-ctranslate2")
    if err := cmd.Run(); err != nil {
        return fmt.Errorf("pip install failed: %w", err)
    }
    
    logger.LogInfo(log, logBuffer, logMutex, "FasterWhisper installed successfully")
    return nil
}
```

#### オプション2: インストール検証の追加
インストール後に `requests` モジュールの存在を確認：

```go
func verifyFasterWhisperInstallation() error {
    // requests モジュールの確認
    cmd := createCommand("python", "-c", "import requests; import faster_whisper")
    if err := cmd.Run(); err != nil {
        return fmt.Errorf("依存パッケージが不足しています: %w", err)
    }
    return nil
}
```

### 再現手順
1. Python 3.12をインストール（pip未更新の状態）
2. KoeMoji-Go v1.8.1を起動
3. faster-whisperが自動インストールされる
4. 音声ファイルを文字起こし実行
5. `ModuleNotFoundError: No module named 'requests'` エラーが発生

### 影響範囲
- Windows環境のみ（macOS/Linuxでは未確認）
- 初回セットアップ時のユーザー体験に影響
- ユーザーが手動でコマンド実行する必要がある

### 優先度
**Medium** - 初回セットアップ時のみ発生、回避策は明確

### 関連ファイル
- `internal/whisper/whisper.go:136-144` (installFasterWhisper関数)

### 参考情報
- faster-whisper dependencies: https://github.com/SYSTRAN/faster-whisper
- requests package: https://pypi.org/project/requests/

--- a/internal/whisper/whisper.go
+++ b/internal/whisper/whisper.go
@@ -134,11 +134,24 @@ func IsFasterWhisperAvailableForTesting() bool {
 }
 
 func installFasterWhisper(log *log.Logger, logBuffer *[]logger.LogEntry, logMutex *sync.RWMutex) error {
-	logger.LogInfo(log, logBuffer, logMutex, "Installing faster-whisper and whisper-ctranslate2...")
-	cmd := createCommand("pip", "install", "faster-whisper", "whisper-ctranslate2")
+	// Step 1: Upgrade pip to ensure proper dependency resolution
+	logger.LogInfo(log, logBuffer, logMutex, "Upgrading pip to latest version...")
+	pipUpgradeCmd := createCommand("python", "-m", "pip", "install", "--upgrade", "pip")
+	if err := pipUpgradeCmd.Run(); err != nil {
+		logger.LogError(log, logBuffer, logMutex, "pip upgrade failed (non-fatal): %v", err)
+		// Continue even if pip upgrade fails
+	}
+
+	// Step 2: Install faster-whisper with explicit dependencies
+	// Include 'requests' explicitly to avoid indirect dependency resolution issues
+	// Background: requests is an indirect dependency of huggingface-hub (used by faster-whisper)
+	// Older pip versions or certain environments may fail to resolve it automatically
+	logger.LogInfo(log, logBuffer, logMutex, "Installing faster-whisper and dependencies...")
+	cmd := createCommand("pip", "install", "requests", "faster-whisper", "whisper-ctranslate2")
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("pip install failed: %w", err)
 	}
+
 	logger.LogInfo(log, logBuffer, logMutex, "FasterWhisper installed successfully")
 	return nil
 }

